activesupport/lib/active_support/cache/file_store.rb

```diff
@@ -74,7 +74,8 @@ def delete_matched(matcher, options = nil)
       private
         def read_entry(key, **options)
           if File.exist?(key)
-            File.open(key) { |f| Marshal.load(f) }
+            entry = File.open(key) { |f| Marshal.load(f) }
+            entry if entry.respond_to?(:expired?)
```

Comment:
Actually it works, but `entry` is not always called by `expired?` but also `mismatched?`, more generic ways is preferable.

https://github.com/rails/rails/blob/e4b6c719cd1df3dd2bf16645dbe7780dd541ec78/activesupport/lib/active_support/cache.rb#L324-L326

https://github.com/rails/rails/blob/e4b6c719cd1df3dd2bf16645dbe7780dd541ec78/activesupport/lib/active_support/cache.rb#L501-L502

Reply:
>more generic ways is preferable.

I see.
FileStore assumes entry object is a `Cache::Entry` obejct.
I added `is_a?` class checking in `read_etnry` method.